cobalt/media/base/metrics_provider.cc

```diff
@@ -0,0 +1,159 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/media/base/metrics_provider.h"
+
+#include "base/logging.h"
+#include "base/metrics/histogram_functions.h"
+
+
+namespace cobalt {
+namespace media {
+
+using starboard::ScopedLock;
+
+
+MediaMetricsProvider::~MediaMetricsProvider() {
+  if (!IsInitialized()) return;
+  ReportPipelineUMA();
+}
+
+std::string MediaMetricsProvider::GetUMANameForAVStream(
+    const PipelineInfo& player_info) {
+  constexpr char kPipelineUmaPrefix[] = "Media.PipelineStatus.AudioVideo.";
+  std::string uma_name = kPipelineUmaPrefix;
+  if (player_info.video_codec == VideoCodec::kVP8)
+    uma_name += "VP8.";
+  else if (player_info.video_codec == VideoCodec::kVP9)
+    uma_name += "VP9.";
+  else if (player_info.video_codec == VideoCodec::kH264)
+    uma_name += "H264.";
+  else if (player_info.video_codec == VideoCodec::kAV1)
+    uma_name += "AV1.";
+  else
+    return uma_name + "Other";
```

Comment:
Do we need a trailing "." here?

Reply:
I've removed it now. If we're not appending the Pipeline token, we don't need the trailing period.